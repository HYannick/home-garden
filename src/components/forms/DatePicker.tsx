/** @jsx jsx */
import React, { Fragment, useState } from 'react';
import styled from '@emotion/styled';
import { jsx } from '@emotion/core';
import DayPicker, { DayModifiers } from 'react-day-picker';
import { Label } from './Label';
import '../../static/date-picker.styles.css';

interface DateProps {
  field?: any,
  label?: string,
  onDateSelected: any
}

const StyledDayPicker = styled(DayPicker)`
    font-size: 1.5rem;
    outline: white;
    width: 100%;
    border: 0.5rem solid ${({ theme }) => theme.palette.primary.light};
    border-radius: 1rem;
    padding: 1rem;
`;

const DatePicker: React.FC<DateProps> = ({ field, label, onDateSelected }) => {
  const [selectedDay, setSelectedDay] = useState<Date | undefined>(undefined);

  const handleDateClick = (day: Date, { selected }: DayModifiers) => {
    if (selected) {
      setSelectedDay(undefined);
      onDateSelected('');
      return;
    }
    setSelectedDay(day);
    onDateSelected(day);
  };

  return (
    <Fragment>
      <Label htmlFor={field.name}>{label}</Label>
      <StyledDayPicker disabledDays={{ before: new Date() }} onDayClick={handleDateClick} selectedDays={selectedDay}/>
    </Fragment>
  );
};

export default DatePicker;
