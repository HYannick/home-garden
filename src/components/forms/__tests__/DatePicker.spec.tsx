import React from 'react';
import 'jest-dom/extend-expect';
import { cleanup, fireEvent } from '@testing-library/react';
import { renderWithTheme } from '../../../theme-wrapper';
import DatePicker from '../DatePicker';

afterEach(cleanup);
describe('DatePicker', () => {
  const props = {
    field: {
      name: 'puke',
    },
    type: 'text',
    label: 'Should I puke?',
    onDateSelected: jest.fn()
  };
  it('should render properly', () => {
    const { container } = renderWithTheme(
      <DatePicker {...props}/>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should render the right label', () => {
    const { getByText } = renderWithTheme(
      <DatePicker {...props}/>,
    );
    expect(getByText(/^Should I puke/)).toHaveTextContent('Should I puke');
  });
  it('should trigger onChange', () => {
    const { getByText } = renderWithTheme(
      <DatePicker {...props}/>,
    );
    const dateBtn = getByText('17');
    fireEvent.click(dateBtn);
    expect(props.onDateSelected).toHaveBeenCalledTimes(1);
  });
});
