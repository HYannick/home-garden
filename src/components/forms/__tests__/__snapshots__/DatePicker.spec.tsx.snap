// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render properly 1`] = `
<div>
  <label
    class="css-npa6bc"
    for="puke"
  >
    Should I puke?
  </label>
  <div
    class="DayPicker css-1bwhhw4"
    lang="en"
  >
    <div
      class="DayPicker-wrapper"
      tabindex="0"
    >
      <div
        class="DayPicker-NavBar"
      >
        <span
          aria-label="Previous Month"
          class="DayPicker-NavButton DayPicker-NavButton--prev"
          role="button"
          tabindex="0"
        />
        <span
          aria-label="Next Month"
          class="DayPicker-NavButton DayPicker-NavButton--next"
          role="button"
          tabindex="0"
        />
      </div>
      <div
        class="DayPicker-Months"
      >
        <div
          class="DayPicker-Month"
          role="grid"
        >
          <div
            class="DayPicker-Caption"
            role="heading"
          >
            <div>
              July 2019
            </div>
          </div>
          <div
            class="DayPicker-Weekdays"
            role="rowgroup"
          >
            <div
              class="DayPicker-WeekdaysRow"
              role="row"
            >
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Sunday"
                >
                  Su
                </abbr>
              </div>
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Monday"
                >
                  Mo
                </abbr>
              </div>
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Tuesday"
                >
                  Tu
                </abbr>
              </div>
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Wednesday"
                >
                  We
                </abbr>
              </div>
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Thursday"
                >
                  Th
                </abbr>
              </div>
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Friday"
                >
                  Fr
                </abbr>
              </div>
              <div
                class="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Saturday"
                >
                  Sa
                </abbr>
              </div>
            </div>
          </div>
          <div
            class="DayPicker-Body"
            role="rowgroup"
          >
            <div
              class="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled="true"
                class="DayPicker-Day DayPicker-Day--outside"
              />
              <div
                aria-disabled="false"
                aria-label="Mon Jul 01 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="0"
              >
                1
              </div>
              <div
                aria-disabled="false"
                aria-label="Tue Jul 02 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="-1"
              >
                2
              </div>
              <div
                aria-disabled="false"
                aria-label="Wed Jul 03 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="-1"
              >
                3
              </div>
              <div
                aria-disabled="false"
                aria-label="Thu Jul 04 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="-1"
              >
                4
              </div>
              <div
                aria-disabled="false"
                aria-label="Fri Jul 05 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="-1"
              >
                5
              </div>
              <div
                aria-disabled="false"
                aria-label="Sat Jul 06 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="-1"
              >
                6
              </div>
            </div>
            <div
              class="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled="false"
                aria-label="Sun Jul 07 2019"
                aria-selected="false"
                class="DayPicker-Day"
                role="gridcell"
                tabindex="-1"
              >
                7
              </div>
              <div
                aria-disabled="false"
                aria-label="Mon Jul 08 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--today"
                role="gridcell"
                tabindex="-1"
              >
                8
              </div>
              <div
                aria-disabled="true"
                aria-label="Tue Jul 09 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                9
              </div>
              <div
                aria-disabled="true"
                aria-label="Wed Jul 10 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                10
              </div>
              <div
                aria-disabled="true"
                aria-label="Thu Jul 11 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                11
              </div>
              <div
                aria-disabled="true"
                aria-label="Fri Jul 12 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                12
              </div>
              <div
                aria-disabled="true"
                aria-label="Sat Jul 13 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                13
              </div>
            </div>
            <div
              class="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled="true"
                aria-label="Sun Jul 14 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                14
              </div>
              <div
                aria-disabled="true"
                aria-label="Mon Jul 15 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                15
              </div>
              <div
                aria-disabled="true"
                aria-label="Tue Jul 16 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                16
              </div>
              <div
                aria-disabled="true"
                aria-label="Wed Jul 17 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                17
              </div>
              <div
                aria-disabled="true"
                aria-label="Thu Jul 18 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                18
              </div>
              <div
                aria-disabled="true"
                aria-label="Fri Jul 19 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                19
              </div>
              <div
                aria-disabled="true"
                aria-label="Sat Jul 20 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                20
              </div>
            </div>
            <div
              class="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled="true"
                aria-label="Sun Jul 21 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                21
              </div>
              <div
                aria-disabled="true"
                aria-label="Mon Jul 22 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                22
              </div>
              <div
                aria-disabled="true"
                aria-label="Tue Jul 23 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                23
              </div>
              <div
                aria-disabled="true"
                aria-label="Wed Jul 24 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                24
              </div>
              <div
                aria-disabled="true"
                aria-label="Thu Jul 25 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                25
              </div>
              <div
                aria-disabled="true"
                aria-label="Fri Jul 26 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                26
              </div>
              <div
                aria-disabled="true"
                aria-label="Sat Jul 27 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                27
              </div>
            </div>
            <div
              class="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled="true"
                aria-label="Sun Jul 28 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                28
              </div>
              <div
                aria-disabled="true"
                aria-label="Mon Jul 29 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                29
              </div>
              <div
                aria-disabled="true"
                aria-label="Tue Jul 30 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                30
              </div>
              <div
                aria-disabled="true"
                aria-label="Wed Jul 31 2019"
                aria-selected="false"
                class="DayPicker-Day DayPicker-Day--disabled"
                role="gridcell"
                tabindex="-1"
              >
                31
              </div>
              <div
                aria-disabled="true"
                class="DayPicker-Day DayPicker-Day--disabled DayPicker-Day--outside"
              />
              <div
                aria-disabled="true"
                class="DayPicker-Day DayPicker-Day--disabled DayPicker-Day--outside"
              />
              <div
                aria-disabled="true"
                class="DayPicker-Day DayPicker-Day--disabled DayPicker-Day--outside"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
